### PR TITLE
use explore server

### PR DESCRIPTION
**Pull request by etiennehomer**
Signed-off-by: HOMER Etienne <etiennehomer@gmail.com>

### CODE REVIEW

File: src/utils/rest-api.js [hunk 1]
```diff
@@ -16,6 +16,8 @@ let PREFIX_CONFIG_NOTIFICATION_WS =
 let PREFIX_CONFIG_QUERIES = process.env.REACT_APP_API_GATEWAY + '/config';
 let PREFIX_DIRECTORY_SERVER_QUERIES =
     process.env.REACT_APP_API_GATEWAY + '/directory';
+let PREFIX_EXPLORE_SERVER_QUERIES =
```
Comment (sBouzols) [inline src/utils/rest-api.js]: should be `const` variable not `let`, same for others PREFIX_* (in another PR ?)

File: src/utils/rest-api.js [hunk 2]
```diff
@@ -121,7 +123,7 @@ export function fetchDirectoryContent(directoryUuid) {
 export function deleteElement(elementUuid) {
```
Comment (sBouzols) [inline src/utils/rest-api.js:123]: Maybe we should give an order in this file to group REST API calls by service for readability, or even better separate into different files then import them in `rest-api.js`

**Review by sBouzols — CHANGES_REQUESTED** [review]
fine code review
Maybe some refactoring to handle in another US ?

File: src/utils/rest-api.js (hunk 2 above)
Comment (sBouzols) [reply src/utils/rest-api.js:123]: in another US / PR

**Review by FranckLecuyer — COMMENTED** [review]
The functions : 

fetchStudiesInfos
fetchContingencyListsInfos
fetchFiltersInfos

in file rest-api.js

must call the explore-server (which then will call the directory-server and the study-server, actions-sever and filter-server)

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]
The replacement of a filters contingency list with a script contingency list is done in action-server, but the element type is not changed in the directory-server database.
Same problem with filters

File: src/components/directory-content.js [hunk 3]
```diff
@@ -285,16 +288,21 @@ const DirectoryContent = () => {
...
                 setCurrentScriptContingencyListId(event.rowData.elementUuid);
                 setOpenScriptContingencyDialog(true);
-            } else if (event.rowData.type === elementType.SCRIPT) {
+            } else if (
+                event.rowData.type === elementType.FILTER &&
+                subtype === filterSubtype.SCRIPT
+            ) {
                 setCurrentScriptId(event.rowData.elementUuid);
                 setOpenScriptDialog(true);
             } else if (event.rowData.type === elementType.FILTER) {
```
Comment (sBouzols) [inline src/components/directory-content.js:302]: `&&  subtype === filterSubtype.FILTER` is missing here.

File: src/components/directory-content.js [hunk 4]
```diff
@@ -512,11 +520,38 @@ const DirectoryContent = () => {
         return href;
     }
 
+    function buildTypeWithSubtype(type, subtype) {
+        switch (type) {
+            case elementType.FILTER:
+                return subtype === filterSubtype.SCRIPT
+                    ? filterSubtype.SCRIPT
+                    : type;
```
Comment (sBouzols) [inline src/components/directory-content.js]: replace `type` by `filterSubtype.FILTER` to be clearer

File: src/components/directory-content.js [hunk 5]
```diff
@@ -512,11 +520,38 @@ const DirectoryContent = () => {
...
+            case elementType.CONTINGENCY_LIST:
+                return subtype === contingencyListSubtype.FILTERS
+                    ? contingencyListSubtype.FILTERS +
+                          '_' +
+                          elementType.CONTINGENCY_LIST
+                    : contingencyListSubtype.SCRIPT +
```
Comment (sBouzols) [inline src/components/directory-content.js]: Not a fan of this syntax but it works... for now...

File: src/components/directory-content.js [hunk 6]
```diff
@@ -535,17 +570,26 @@ const DirectoryContent = () => {
...
+            objectType === elementType.CONTINGENCY_LIST &&
+            objectSubtype === contingencyListSubtype.FILTERS
+        ) {
             return <PanToolIcon className={classes.icon} />;
+        } else if (
+            objectType === elementType.FILTER &&
+            objectSubtype === filterSubtype.SCRIPT
+        ) {
+            return <FilterIcon className={classes.icon} />;
         } else if (objectType === elementType.FILTER) {
```
Comment (sBouzols) [inline src/components/directory-content.js:573]: `&&  objectSubtype === filterSubtype.FILTER` is missing

File: src/components/directory-content.js [hunk 7]
```diff
@@ -563,10 +607,14 @@ const DirectoryContent = () => {
...
-                {childrenMetadata[elementUuid] && getElementIcon(objectType)}
+                {childrenMetadata[elementUuid] &&
+                    getElementIcon(
+                        objectType,
+                        childrenMetadata[elementUuid].subtype
+                    )}
                 {/* Name */}
                 {isMetadataLoading ? null : childrenMetadata[elementUuid] ? (
-                    <div>{childrenMetadata[elementUuid].name}</div>
+                    <div>{elementName}</div>
```
Comment (sBouzols) [inline src/components/directory-content.js]: revert this, you should always prefer the name from the metadata I think

File: src/components/directory-content.js [hunk 8]
```diff
@@ -774,29 +787,44 @@ const DirectoryContent = () => {
         let children = getSelectedChildren();
         return (
             children.length === 1 &&
-            children[0].type === elementType.FILTERS_CONTINGENCY_LIST
+            children[0].type === elementType.CONTINGENCY_LIST &&
+            childrenMetadata[children[0].elementUuid].subtype ===
```
Comment (sBouzols) [inline src/components/directory-content.js:778]: maybe you should check if `childrenMetadata[children[0].elementUuid]` exists here first.
This is dirty code... OMG...

File: src/components/directory-content.js [hunk 9]
```diff
@@ -774,29 +787,44 @@ const DirectoryContent = () => {
...
         );
     };
 
     const allowsReplaceContingencyWithScript = () => {
         let children = getSelectedChildren();
         return (
             children.length === 1 &&
-            children[0].type === elementType.FILTERS_CONTINGENCY_LIST &&
+            children[0].type === elementType.CONTINGENCY_LIST &&
+            childrenMetadata[children[0].elementUuid].subtype ===
```
Comment (sBouzols) [inline src/components/directory-content.js:789]: same, must check if `childrenMetadata[children[0].elementUuid]` is OK first

File: src/components/directory-content.js [hunk 10]
```diff
@@ -774,29 +787,44 @@ const DirectoryContent = () => {
...
     };
 
     const allowsCopyFilterToScript = () => {
         let children = getSelectedChildren();
-        return children.length === 1 && children[0].type === elementType.FILTER;
+        return (
+            children.length === 1 &&
+            children[0].type === elementType.FILTER &&
+            !(
+                childrenMetadata[children[0].elementUuid].subtype ===
```
Comment (sBouzols) [inline src/components/directory-content.js:802]: same

File: src/components/directory-content.js [hunk 11]
```diff
@@ -774,29 +787,44 @@ const DirectoryContent = () => {
...
+        );
     };
 
     const allowsReplaceFilterWithScript = () => {
         let children = getSelectedChildren();
         return (
             children.length === 1 &&
             children[0].type === elementType.FILTER &&
+            !(
+                childrenMetadata[children[0].elementUuid].subtype ===
```
Comment (sBouzols) [inline src/components/directory-content.js:815]: same

File: src/components/directory-content.js [hunk 12]
```diff
@@ -1087,15 +1115,17 @@ const DirectoryContent = () => {
                 onClose={handleCloseScriptContingency}
                 onError={handleError}
                 title={useIntl().formatMessage({ id: 'editContingencyList' })}
-                type={elementType.SCRIPT_CONTINGENCY_LIST}
+                type={elementType.CONTINGENCY_LIST}
+                subtype={contingencyListSubtype.SCRIPT}
```
Comment (sBouzols) [inline src/components/directory-content.js]: I don't think you need this subtype props here. We check the subtype before setting `openScriptContingencyDialog` then it's obviously `contingencyListSubtype.SCRIPT`

File: src/components/directory-content.js [hunk 13]
```diff
@@ -1087,15 +1115,17 @@ const DirectoryContent = () => {
...
             />
             <ScriptDialog
                 id={currentScriptId}
                 open={openScriptDialog}
                 onClose={handleCloseScriptDialog}
                 onError={handleError}
                 title={useIntl().formatMessage({ id: 'editFilterScript' })}
-                type={elementType.SCRIPT}
+                type={elementType.FILTER}
+                subtype={filterSubtype.SCRIPT}
```
Comment (sBouzols) [inline src/components/directory-content.js]: same remark here

File: src/components/dialogs/script-dialog.js [hunk 14]
```diff
@@ -51,8 +54,9 @@ const useStyles = makeStyles(() => ({
  * @param onError handle errors
  * @param title Title of the dialog
  * @param type Contingencies or filter
+ * @param subtype Element's subtype
  */
-const ScriptDialog = ({ id, open, onClose, onError, title, type }) => {
+const ScriptDialog = ({ id, open, onClose, onError, title, type, subtype }) => {
```
Comment (sBouzols) [inline src/components/dialogs/script-dialog.js]: I don't thnik you need a subtype prop here, the main type is sufficient to manage conditions in this dialog.

**Review by sBouzols — CHANGES_REQUESTED** [review]
Fine code Review
No tests

File: src/utils/rest-api.js (hunk 1 above)
Comment (etiennehomer) [reply src/utils/rest-api.js]: Done in this one.

File: src/components/directory-content.js (hunk 11 above)
Comment (etiennehomer) [reply src/components/directory-content.js:815]: Done

File: src/components/directory-content.js (hunk 10 above)
Comment (etiennehomer) [reply src/components/directory-content.js:802]: done

File: src/components/directory-content.js (hunk 8 above)
Comment (etiennehomer) [reply src/components/directory-content.js:778]: Done.
Yes it's complicated... Maybe we should get the available menu actions from the back.

File: src/components/directory-content.js (hunk 7 above)
Comment (etiennehomer) [reply src/components/directory-content.js]: done

File: src/components/directory-content.js (hunk 3 above)
Comment (etiennehomer) [reply src/components/directory-content.js:302]: Actually, the subtype we get for the filter metadata is SCRIPT or LINE, LOAD, BATTERY etc. So we can't test if subtype === FILTER. It has to be changed in this US or not...

File: src/components/directory-content.js (hunk 4 above)
Comment (etiennehomer) [reply src/components/directory-content.js]: done

File: src/components/directory-content.js (hunk 9 above)
Comment (etiennehomer) [reply src/components/directory-content.js:789]: done

File: src/components/directory-content.js (hunk 12 above)
Comment (etiennehomer) [reply src/components/directory-content.js]: You're right. Done.

File: src/components/directory-content.js (hunk 13 above)
Comment (etiennehomer) [reply src/components/directory-content.js]: done

File: src/components/dialogs/script-dialog.js (hunk 14 above)
Comment (etiennehomer) [reply src/components/dialogs/script-dialog.js]: done

File: src/components/directory-content.js (hunk 8 above)
Comment (sBouzols) [reply src/components/directory-content.js:778]: It's not your dirty code :sweat_smile:

File: src/components/directory-content.js (hunk 3 above)
Comment (etiennehomer) [reply src/components/directory-content.js:302]: Changed was done in filter-server and in grid-explore

**Review by sBouzols — APPROVED** [review]

**Pull request merged**